### PR TITLE
Small simplifications in helper functions.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide ptal.

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs [hunk 1]
```diff
@@ -747,14 +732,8 @@ private static bool IsAllContainingTypeDeclsPublic(SyntaxNode node)
...
             var containingTypeDeclarations = node.GetAncestors<TypeDeclarationSyntax>();
-            if (containingTypeDeclarations.Count() == 0)
-            {
-                return true;
-            }
-            else
-            {
-                return containingTypeDeclarations.All(typedecl => typedecl.GetModifiers().Any(m => m.Kind() == SyntaxKind.PublicKeyword));
-            }
+            return !containingTypeDeclarations.Any()
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs:735]: preserving from before.  not 100% sure.  but didn't want to break anything.

File: src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs [hunk 2]
```diff
@@ -697,43 +697,28 @@ internal override bool IsPublicOnlyAccessibility(ExpressionSyntax expression, Pr
...
-                        {
-                            return IsAllContainingTypeDeclsPublic(typeDecl);
-                        }
-                        else
-                        {
-                            // The Type Decl which contains the BaseList does not contain Public
-                            return false;
-                        }
+                        // The Type Decl which contains the BaseList needs to contain Public
+                        return typeDecl.GetModifiers().Any(SyntaxKind.PublicKeyword) && IsAllContainingTypeDeclsPublic(typeDecl);
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs:707]: It's more readable to me. Wcyd?

**Pull request merged**